### PR TITLE
chore(deps): upgrade python protobuf to latest 3.20 version

### LINKED ISSUE #14314: Update python protobuf to latest 3.20 version
Protobuf has released a new 3.20 patch to handle some security issues.
https://github.com/protocolbuffers/protobuf/security/advisories/GHSA-8gq9-2x98-w8hf

While this was already noticed by [dependabot](https://github.com/magma/magma/pull/14010), it's unfortunately a bit more difficult to merge this:
* Also update the dependencies within Bazel (requirements.in, requirements.txt).
* Make sure pydep uses the upgrade (look at the lockfile).
* Run debian integration tests.

Tasks:
* [x] Update protobuf dependency to latest 3.20.x version in setup.py, requirements.in, requirements.txt
* [x] Build a Magma debian release and make sure that pydep actually uses the new version.
* [x] Run debian integration tests.
* [x] Monitor the CI for build issues after merging.
* [x] Prepare internal review meeting.

### PR DESCRIPTION
**Pull request by mpfirrmann**
Signed-off-by: Marco Pfirrmann <marco.pfirrmann@tngtech.com>

## Summary

This protobuf version patches [a potential Denial of Service issue](https://github.com/protocolbuffers/protobuf/security/advisories/GHSA-8gq9-2x98-w8hf). Based on #14388.
Closes #14314.

## Test Plan

Tested locally:

1. Build new `magma.deb` with `fab release package`.
2. Uninstall existing `protobuf 3.20.3` package with `pip3 uninstall protobuf`.
3. Install created `*.deb` packages.
4. Prepare VM for testing with the steps described in `magma_deb/tasks/main.yml`.
5. Spin up `magma_test` and `magma_trfserver` and run S1AP integration tests.

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

File: bazel/external/requirements.in [hunk 1]
```diff
@@ -3,7 +3,7 @@ setuptools==49.6.0
 grpcio-tools<1.49.0
 grpcio<1.49.0
 redis>=3.5
-protobuf
+protobuf==3.20.2
```
Comment (sebathomas) [inline bazel/external/requirements.in]: There's a newer version 3.20.3 now.

Comment (mpfirrmann) [reply bazel/external/requirements.in]: Sorry about that. Changed.

**Review by sebathomas — COMMENTED** [review]
What about the updated pydep lockfile?

File: bazel/external/requirements.in [hunk 2]
```diff
@@ -3,7 +3,7 @@ setuptools==49.6.0
 grpcio-tools<1.49.0
 grpcio<1.49.0
 redis>=3.5
-protobuf
+protobuf==3.20.3
```
Comment (nstng) [inline bazel/external/requirements.in]: Should we use ">="? It was not pinned before.

**Comment by mpfirrmann** [comment]
> What about the updated pydep lockfile?

I thought this would be created automatically with the next `magma.deb`, will recreate and add it.

**Review by sebathomas — APPROVED** [review]

**Review by nstng — APPROVED** [review]
bazel changes lgtm

**Pull request merged**